concurrent/src/test/java/com/facebook/airlift/concurrent/TestConcurrentScheduledExecutor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.concurrent;
+
+import org.testng.annotations.Test;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestConcurrentScheduledExecutor
+{
+    @Test
+    public void testExecute()
+            throws Exception
+    {
+        ConcurrentScheduledExecutor scheduler = new ConcurrentScheduledExecutor(2, 2, "test-");
+        AtomicInteger counter = new AtomicInteger();
+        scheduler.execute(() -> {
+            counter.incrementAndGet();
+        });
+        scheduler.awaitTermination(1, TimeUnit.SECONDS);
+        assertEquals(counter.get(), 1);
+        scheduler.shutdownNow();
```

Comment:
Maybe let's do `assertTrue(scheduler.awaitTermination(10, TimeUnit.SECONDS))` at the end of each test to avoid thread leaks